### PR TITLE
Fix backup download by name

### LINKED ISSUE #59: backup download is broken when using name (instead of ID)
```
crowbarctl backup download mybackup
/usr/lib64/ruby/gems/2.1.0/gems/crowbar-client-2.3.0/lib/crowbar/client/command/backup/download.rb:61:in `path': undefined method `[]' for nil:NilClass (NoMethodError)
        from /usr/lib64/ruby/gems/2.1.0/gems/crowbar-client-2.3.0/lib/crowbar/client/command/backup/download.rb:37:in `block in execute'
        from /usr/lib64/ruby/gems/2.1.0/gems/crowbar-client-2.3.0/lib/crowbar/client/request/base.rb:75:in `process'
        from /usr/lib64/ruby/gems/2.1.0/gems/crowbar-client-2.3.0/lib/crowbar/client/command/backup/download.rb:31:in `execute'
        from /usr/lib64/ruby/gems/2.1.0/gems/crowbar-client-2.3.0/lib/crowbar/client/app/backup.rb:105:in `download'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:115:in `invoke'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:235:in `block in subcommand'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from /usr/lib64/ruby/gems/2.1.0/gems/crowbar-client-2.3.0/lib/crowbar/client/util/runner.rb:47:in `execute!'
        from /usr/lib64/ruby/gems/2.1.0/gems/crowbar-client-2.3.0/bin/crowbarctl:34:in `<top (required)>'
        from /usr/bin/crowbarctl:23:in `load'
        from /usr/bin/crowbarctl:23:in `<main>'
```

### PR DESCRIPTION
**Pull request by rsalevsky**
Fixes #59

### CODE REVIEW

**Comment by dirkmueller** [comment]
Hopefully backups never have the id "0"..

**Comment by dirkmueller** [comment]
+1

**Comment by MaximilianMeister** [comment]
any idea why travis fails?

**Comment by rsalevsky** [comment]
@MaximilianMeister Nop, also @tboerger has no idea. But it's not regarding to this PR I would say.

**Comment by MaximilianMeister** [comment]
if you are sure that this PR didn't introduce the failure +1

File: lib/crowbar/client/command/backup/download.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@ def write(body)
 
           def path
             filename = Request::Backup::List.new.process do |p|
-              p.parsed_response.select { |s| s["id"] == args.id.to_i }
+              p.parsed_response.select { |s| s["id"] == args.id.to_i || s["name"] == args.id }
```
Comment (tboerger) [inline lib/crowbar/client/command/backup/download.rb]: Please use `detect` instead of `select`, that directly returns the first matching row.

**Comment by tboerger** [comment]
Obsolete in favor of https://github.com/crowbar/crowbar-client/pull/66

**Pull request closed without merging**